### PR TITLE
Facility samplers

### PR DESCRIPTION
**Pull request by Theodore-Chatziioannou**
Adds the ability to weight on a specified facility attribute (such as floorspace/units/floors etc) when building a facility sampler. 
This functionality can be added with relatively small changes in the existing sampling approach, so I am introducing it as a separate pull request to the full-version, which will follow later and will be implemented as a new module.

### CODE REVIEW

**Review by fredshone — APPROVED** [review]
Great.

This will work with the exitsting format (from osmfs). But maybe we update when i provide some test data for the new 'osm extractor'.

**Review by KasiaKoz — APPROVED** [review]
looks good

File: pam/samplers/facility.py [hunk 1]
```diff
@@ -121,7 +127,8 @@ def build_facilities_sampler(self, facilities, zones):
                 facs = zone_facs.loc[zone_facs.activity == act]
                 if not facs.empty:
                     points = [(i, g) for i, g in facs.geometry.items()]
-                    sampler_dict[zone][act] = inf_yielder(points)
+                    weights = facs[weight_on] / facs[weight_on].sum() if weight_on != None else None # sum of weights/probabilities should be one
```
[the remark this answers is not in the document]
Comment (Theodore-Chatziioannou) [reply pam/samplers/facility.py]: good point - split it over two lines now :)

**Comment by Theodore-Chatziioannou** [comment]
> Great.
> 
> This will work with the exitsting format (from osmfs). But maybe we update when i provide some test data for the new 'osm extractor'.

@fredshone  The reason I used geodf columns as weights instead of the nested json format is the easier manipulation/combination of weights. I suspect a user would want to combine fields like area/units/floors/transit/activities to calculate the probabilities in each case, which is easier with a geodataframe format, done between the osmfs and pam steps. But we can revisit in the future.

**Pull request merged**